rust/models/src/credential.rs

```diff
@@ -114,6 +121,14 @@ pub enum CredentialType {
         #[cfg_attr(feature = "serde_support", serde(rename = "private"))]
         /// The private key for authentication.
         private_key: String,
+        /// privilege username only use for SSH service
+        #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
This is a personal preference so it can be seen as nitpicking and can be decided by you. But I think that keeping the information as null within the response is better than hiding it as it makes it more obvious that it is an option without having to lookup the api when playing around with openvasd. I am aware that some prefer it, and I have to admit I was one of them, in this case I think it is not necessary to spare 50 bytes.

Reply:
TL;DR: I  would remove `#[serde(skip_serializing_if = "Option::is_none")]`

Reply:
I thought about this as well, but it is not only for SSH service. The problem is that it will be shown for all service credential, even for those which does not allow/support privilege credentials. 
I decide to hide it because it can be confusing an unset privilege credential for e.g. SMB service, which doesn't support it

Reply:
that's a good point.